src/util.rs

```diff
@@ -19,20 +35,44 @@ impl LazyUsize {
 
     // The initialization is not completed.
     pub const UNINIT: usize = usize::max_value();
+    // The initialization is currently running.
+    pub const ACTIVE: usize = usize::max_value() - 1;
 
     // Runs the init() function at least once, returning the value of some run
-    // of init(). Unlike std::sync::Once, the init() function may be run
-    // multiple times. If init() returns UNINIT, future calls to unsync_init()
-    // will always retry. This makes UNINIT ideal for representing failure.
+    // of init(). Multiple callers can run their init() functions in parallel.
+    // init() should always return the same value, if it succeeds.
     pub fn unsync_init(&self, init: impl FnOnce() -> usize) -> usize {
         // Relaxed ordering is fine, as we only have a single atomic variable.
```

Comment:
This assumes that a file descriptor consists *only* of that integer and no other user-space data. I suppose that is accurate on Unixes?

Reply:
Yup this is correct, [`libc::open` always returns a positive int](http://man7.org/linux/man-pages/man2/open.2.html) (link is to the Linux docs, but the function is a POSIX standard).

So a file descriptor will always fit in a positive int on unix